resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/submit/v2/MountedDependencyManagerImpl.scala

```diff
@@ -0,0 +1,283 @@
+    }.toMap
+    val jarsSecretBase64 = BaseEncoding.base64().encode(jarsSecret.getBytes(Charsets.UTF_8))
+    val filesSecretBase64 = BaseEncoding.base64().encode(filesSecret.getBytes(Charsets.UTF_8))
+    val secretData = Map(
+      INIT_CONTAINER_DOWNLOAD_JARS_SECRET_KEY -> jarsSecretBase64,
+      INIT_CONTAINER_DOWNLOAD_FILES_SECRET_KEY -> filesSecretBase64) ++
+      trustStoreBase64
+    val kubernetesSecret = new SecretBuilder()
+      .withNewMetadata()
+      .withName(s"$kubernetesAppId-spark-init")
+      .endMetadata()
+      .addToData(secretData.asJava)
+      .build()
+    kubernetesSecret
+  }
+
+  override def buildInitContainerConfigMap(): ConfigMap = {
+    val initContainerProperties = new Properties()
+    initContainerProperties.setProperty(RESOURCE_STAGING_SERVER_URI.key, stagingServerUri)
+    initContainerProperties.setProperty(DRIVER_LOCAL_JARS_DOWNLOAD_LOCATION.key, jarsDownloadPath)
+    initContainerProperties.setProperty(DRIVER_LOCAL_FILES_DOWNLOAD_LOCATION.key, filesDownloadPath)
+    initContainerProperties.setProperty(
+      INIT_CONTAINER_DOWNLOAD_JARS_SECRET_LOCATION.key, INIT_CONTAINER_DOWNLOAD_JARS_SECRET_PATH)
+    initContainerProperties.setProperty(
+      INIT_CONTAINER_DOWNLOAD_FILES_SECRET_LOCATION.key, INIT_CONTAINER_DOWNLOAD_FILES_SECRET_PATH)
+    initContainerProperties.setProperty(DRIVER_MOUNT_DEPENDENCIES_INIT_TIMEOUT.key,
+      s"${downloadTimeoutMinutes}m")
+    stagingServiceSslOptions.trustStore.foreach { _ =>
+      initContainerProperties.setProperty(RESOURCE_STAGING_SERVER_TRUSTSTORE_FILE.key,
+        INIT_CONTAINER_TRUSTSTORE_PATH)
+    }
+    initContainerProperties.setProperty(RESOURCE_STAGING_SERVER_SSL_ENABLED.key,
+      stagingServiceSslOptions.enabled.toString)
+    stagingServiceSslOptions.trustStorePassword.foreach { password =>
+      initContainerProperties.setProperty(RESOURCE_STAGING_SERVER_TRUSTSTORE_PASSWORD.key, password)
+    }
+    stagingServiceSslOptions.trustStoreType.foreach { storeType =>
+      initContainerProperties.setProperty(RESOURCE_STAGING_SERVER_TRUSTSTORE_TYPE.key, storeType)
+    }
+    val propertiesWriter = new StringWriter()
+    initContainerProperties.store(propertiesWriter, "Init-container properties.")
+    new ConfigMapBuilder()
+      .withNewMetadata()
+      .withName(s"$kubernetesAppId-init-properties")
+      .endMetadata()
+      .addToData(INIT_CONTAINER_CONFIG_MAP_KEY, propertiesWriter.toString)
+      .build()
+  }
+
+  override def resolveSparkJars(): Seq[String] = resolveLocalFiles(sparkJars, jarsDownloadPath)
+
+  override def resolveSparkFiles(): Seq[String] = resolveLocalFiles(sparkFiles, filesDownloadPath)
+
+  private def resolveLocalFiles(allFiles: Seq[String], localDownloadRoot: String): Seq[String] = {
+    allFiles.map { file =>
+      Option(Utils.resolveURI(file).getScheme).getOrElse("file") match {
+        case "file" =>
+          val fileName = new File(file).getName
+          s"file://$localDownloadRoot/$fileName"
```

Comment:
In the upload we rename the files with deduplication counters - we should probably do the same thing here.